hospital/src/test/java/by/emel/anton/controller/DoctorControllerTest.java

```diff
@@ -0,0 +1,183 @@
+    private static final String PARAM_PAT_ID = "patientId";
+    private static final String DEFAULT_DOCTOR_NAME = "defaultDoctorName";
+    private static final String DEFAULT_DOCTOR_LOGIN = "defaultDoctorLogin";
+    private static final String DEFAULT_DOCTOR_PASS = "defaultPass";
+    private static final String DEFAULT_PATIENT_NAME = "defaultPatientName";
+    private static final String DEFAULT_PATIENT_LOGIN = "defaultPatientLogin";
+    private static final String DEFAULT_PATIENT_PASS = "defaultPatientPass";
+
+    private MockMvc mockMvc;
+    private ObjectMapper objectMapper;
+    private UserJpaRepository userJpaRepository;
+    private DoctorJpaRepository doctorJpaRepository;
+    private PatientJpaRepository patientJpaRepository;
+    private TherapyJpaRepository therapyJpaRepository;
+
+    @Autowired
+    public DoctorControllerTest(
+            MockMvc mockMvc,
+            ObjectMapper objectMapper,
+            UserJpaRepository userJpaRepository,
+            TherapyJpaRepository therapyJpaRepository,
+            DoctorJpaRepository doctorJpaRepository,
+            PatientJpaRepository patientJpaRepository) {
+        this.mockMvc = mockMvc;
+        this.objectMapper = objectMapper;
+        this.userJpaRepository = userJpaRepository;
+        this.therapyJpaRepository = therapyJpaRepository;
+        this.doctorJpaRepository = doctorJpaRepository;
+        this.patientJpaRepository = patientJpaRepository;
+    }
+
+    @BeforeEach
+    private void fillInTestDataBase() {
+        userJpaRepository.deleteAll();
+        therapyJpaRepository.deleteAll();
+
+        Doctor doctor = new Doctor();
+        doctor.setLogin(DEFAULT_DOCTOR_LOGIN);
+        doctor.setPassword(DEFAULT_DOCTOR_PASS);
+        doctor.setName(DEFAULT_DOCTOR_NAME);
+        doctor.setBirthday(LocalDate.of(1901, 1, 1));
+        doctor.setPatients(new ArrayList<>());
+
+        Patient patient = new Patient();
+        patient.setLogin(DEFAULT_PATIENT_LOGIN);
+        patient.setPassword(DEFAULT_PATIENT_PASS);
+        patient.setName(DEFAULT_PATIENT_NAME);
+        patient.setBirthday(LocalDate.of(1990, 1, 1));
+        patient.setTherapies(new ArrayList<>());
+        doctor.addPatient(patient);
+        patient.setDoctor(doctor);
+        userJpaRepository.save(doctor);
+        userJpaRepository.save(patient);
+    }
+
+    @Test
+    public void getDoctorByLoginPass() throws Exception {
+        Doctor doctor = doctorJpaRepository.findAll().stream().findFirst().orElseThrow(Exception::new);
+        ResponseDoctorDTO responseDoctorDTO = new ResponseDoctorDTO();
+        responseDoctorDTO.setId(doctor.getId());
+        responseDoctorDTO.setPatientIds(doctor.getPatients().stream().map(User::getId).collect(Collectors.toList()));
+        responseDoctorDTO.setName(doctor.getName());
+        responseDoctorDTO.setLogin(doctor.getLogin());
+        String jsonResponseBody = objectMapper.writeValueAsString(responseDoctorDTO);
```

Comment:
I will suggest to use as little production code as possible. Good idea is to have all requests and responses as a separate text files under test/resource fodler. First of all you test that expected json body converts properly to your dto. Also, this files could be a good documentation (example) of what do you expect as a request body. The same applies for response.